File: holoeasy-core/src/main/java/org/holoeasy/animation/Animations.java
package org.holoeasy.animation;

import com.github.retrooper.packetevents.PacketEvents;
import com.github.retrooper.packetevents.wrapper.play.server.WrapperPlayServerEntityRotation;
import org.bukkit.entity.Player;
import org.bukkit.scheduler.BukkitRunnable;
import org.bukkit.scheduler.BukkitTask;
import org.holoeasy.hologram.Hologram;
import org.holoeasy.line.LineImpl;

import java.util.function.Function;

public enum Animations {

    CIRCLE(line -> {
        Hologram holo = line.getHologram();

        return new BukkitRunnable() {

            float yaw = 0.0f;
            @Override
            public void run() {
                for (Player seeingPlayer : holo.getPvt().getSeeingPlayers()) {
                    WrapperPlayServerEntityRotation packet = new WrapperPlayServerEntityRotation(line.getEntityID(), (float) yaw, 0f, false);
                    PacketEvents.getAPI().getPlayerManager().sendPacket(seeingPlayer, packet);
                }
            }
        }.runTaskTimerAsynchronously(holo.getLib().getPlugin(), 2, 2);
    });


    private final Function<LineImpl<?>, BukkitTask> task;


    Animations(Function<LineImpl<?>, BukkitTask> task) {
        this.task = task;
    }

    public Function<LineImpl<?>, BukkitTask> getTask() {
        return task;
    }
}

File: holoeasy-core/src/main/kotlin/org/holoeasy/animation/Animations.kt
package org.holoeasy.animation

import org.bukkit.scheduler.BukkitTask
import org.holoeasy.HoloEasy
import org.holoeasy.line.LineImpl
import org.holoeasy.util.BukkitFuture

enum class Animations(val task: (LineImpl<*>) -> BukkitTask) {

    CIRCLE({ line ->
        val holo = line.hologram
        var yaw = 0.0

        BukkitFuture.runTaskTimerAsynchronously(holo.lib.plugin, 2, 2) {
            holo.pvt.seeingPlayers.forEach { player ->
                holo.lib.packetImpl.rotate(player, line.entityID, yaw = yaw)
            }

            yaw += 10
        }

    })
    ;


}